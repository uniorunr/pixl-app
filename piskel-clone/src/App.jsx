import React, { Component, Fragment } from 'react';
import ReactDOM from 'react-dom';
import NavBar from './components/NavBar/NavBar';
import Main from './components/Main/Main';
import './App.scss';

class App extends Component {
  constructor() {
    super();

    this.state = {};
  }

  render() {
    return (
      <Fragment>
        <NavBar />
        <Main />
      </Fragment>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('root'));
