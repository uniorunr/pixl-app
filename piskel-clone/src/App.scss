@import url('https://fonts.googleapis.com/css?family=Roboto:400,500,700&display=swap');

html {
  height: 100%;

  body {
    height: 100%;
    margin: 0;
    font-size: 10px;
    font-family: "Roboto", "Arial", sans-serif;
  }
}


