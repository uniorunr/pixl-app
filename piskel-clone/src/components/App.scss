@import url('https://fonts.googleapis.com/css?family=Roboto:400,500,700&display=swap');

::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  background: #f1f1f1;
}

::-webkit-scrollbar-thumb {
  background: #101010;
}

::-webkit-scrollbar-thumb:hover {
  background: black;
}

html {
  height: 100%;
  font-size: 10px;

  body {
    height: 100%;
    margin: 0;
    font-family: "Roboto", "Arial", sans-serif;
    display: flex;
    flex-direction: column;
    overflow: hidden;

    .root {
      display: flex;
      flex-direction: column;
      overflow: hidden;
    }

    .visually-hidden {
      position: absolute;

      width: 1px;
      height: 1px;
      margin: -1px;

      clip: rect(0 0 0 0);
    }
  }
}


