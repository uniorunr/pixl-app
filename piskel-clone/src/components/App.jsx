import React, { Component, Fragment } from 'react';
import * as firebase from 'firebase/app';
import 'firebase/auth';
import NavBar from './NavBar/NavBar';
import Main from './Main/Main';
import './App.scss';
import FireBase from '../firebase/firebase';
import toolsDataJSON from './toolsData.json';

FireBase.init();

class App extends Component {
  state = {
    userData: null,
    signInState: null,
    toolsData:
      JSON.parse(sessionStorage.getItem('toolsData'))
      || JSON.parse(JSON.stringify(toolsDataJSON)),
    currToolId: 'pen',
  };

  updateCurrentTool = (tool) => {
    this.setState({
      currToolId: tool,
    });
  };

  updateToolsData = (data) => {
    this.setState({
      toolsData: data,
    });
  };

  componentDidMount = () => {
    firebase.auth().onAuthStateChanged((user) => {
      if (user) {
        this.setState({
          userData: user,
          signInState: null,
        });
      }
    });
  };

  updateSignInState = (state) => {
    this.setState({
      signInState: state,
    });
  };

  render() {
    const {
      userData, signInState, toolsData, currToolId,
    } = this.state;

    return (
      <Fragment>
        <NavBar
          userData={userData}
          updateSignInState={this.updateSignInState}
          signInState={signInState}
          toolsData={toolsData}
          updateCurrentTool={this.updateCurrentTool}
          updateToolsData={this.updateToolsData}
        />
        <Main
          toolsData={toolsData}
          currToolId={currToolId}
          updateCurrentTool={this.updateCurrentTool}
        />
      </Fragment>
    );
  }
}

export default App;
