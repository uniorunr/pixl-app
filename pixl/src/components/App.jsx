import React, { Component, Fragment } from 'react';
import * as firebase from 'firebase/app';
import 'firebase/auth';
import NavBar from './NavBar/NavBar';
import Main from './Main/Main';
import LandingPage from './LandingPage/LandingPage';
import './App.scss';
import FireBase from '../firebase/firebase';
import appDataJSON from './appData.json';

FireBase.init();

class App extends Component {
  state = {
    userData: null,
    signInState: null,
    section: sessionStorage.getItem('section') || 'landing',
    toolsData:
      JSON.parse(sessionStorage.getItem('toolsData'))
      || JSON.parse(JSON.stringify(appDataJSON)).tools,
    currToolId: 'pen',
    framesShortcuts:
      JSON.parse(sessionStorage.getItem('framesShortcuts'))
      || JSON.parse(JSON.stringify(appDataJSON)).frames,
    layersShortcuts:
      JSON.parse(sessionStorage.getItem('layersShortcuts'))
      || JSON.parse(JSON.stringify(appDataJSON)).layers,
  };

  toggleSection = (state) => {
    this.setState({
      section: state,
    });
  };

  updateCurrentTool = (tool) => {
    this.setState({
      currToolId: tool,
    });
  };

  updateToolsData = (data) => {
    this.setState({
      toolsData: data,
    });
  };

  updateFrameShortcuts = (data) => {
    this.setState({
      framesShortcuts: data,
    });
  };

  updateLayersShortcuts = (data) => {
    this.setState({
      layersShortcuts: data,
    });
  };

  componentDidMount = () => {
    firebase.auth().onAuthStateChanged((user) => {
      if (user) {
        this.setState({
          userData: user,
          signInState: null,
        });
      }
    });
  };

  updateSignInState = (state) => {
    this.setState({
      signInState: state,
    });
  };

  render() {
    const {
      userData,
      signInState,
      toolsData,
      currToolId,
      framesShortcuts,
      layersShortcuts,
      section,
    } = this.state;

    return (
      <Fragment>
        <NavBar
          userData={userData}
          updateSignInState={this.updateSignInState}
          signInState={signInState}
          toolsData={toolsData}
          framesShortcuts={framesShortcuts}
          layersShortcuts={layersShortcuts}
          updateCurrentTool={this.updateCurrentTool}
          updateToolsData={this.updateToolsData}
          updateFrameShortcuts={this.updateFrameShortcuts}
          updateLayersShortcuts={this.updateLayersShortcuts}
          toggleSection={this.toggleSection}
        />
        {section === 'landing' ? (
          <LandingPage toggleSection={this.toggleSection} />
        ) : (
          <Main
            toolsData={toolsData}
            currToolId={currToolId}
            updateCurrentTool={this.updateCurrentTool}
            framesShortcuts={framesShortcuts}
            layersShortcuts={layersShortcuts}
          />
        )}
      </Fragment>
    );
  }
}

export default App;
