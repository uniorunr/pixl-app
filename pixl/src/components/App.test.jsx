import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { JSDOM } from 'jsdom';
import App from './App';
import store from '../stores/mainStore';

const dom = new JSDOM();
global.document = dom.window.document;
global.window = dom.window;

function createNodeMock(element) {
  if (typeof element.type === 'string' && element.type === 'div') {
    const elm = document.createElement(element.type);
    elm.setAttribute('data-react-beautiful-dnd-drag-handle', '0');
    return elm;
  }
  if (element.type === 'input') {
    return {
      current: {
        value: 23,
      },
    };
  }
  return null;
}

const canvas = document.createElement('canvas');
canvas.setAttribute('id', 'canvas');
document.body.appendChild(canvas);

it('renders correctly', () => {
  const options = { createNodeMock };
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>,
      options,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
