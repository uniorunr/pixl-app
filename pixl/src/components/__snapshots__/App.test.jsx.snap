// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="navbar"
  >
    <nav
      className="navbar__content"
    >
      <h1
        className="navbar__logo"
        onClick={[Function]}
        role="presentation"
      >
        <a
          href="/"
        >
          <img
            alt="Pixl logo"
            src="test-file-stub"
          />
          <span>
            Pixl!
          </span>
        </a>
      </h1>
      <div
        className="navbar__info"
      >
        <button
          className="navbar__faq"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-question"
          />
        </button>
        <div
          className="navbar__tooltip"
        >
          <span>
            Info
          </span>
        </div>
      </div>
      <button
        className="navbar__sign-in-button"
        onClick={[Function]}
        type="button"
      >
        <span>
          Sign in
        </span>
      </button>
    </nav>
  </div>,
  <main
    className="landing-page"
  >
    <div
      className="landing-page__content"
    >
      <div
        className="landing-page__left-section"
      >
        <div
          className="landing-page__description"
        >
          <h2>
            PIXL! - Pixel Art Tool
          </h2>
          <p>
            Create animation in just one step. Almost all kind of tools for drawing, Frame management, Layer management, Primary and secondary color, Configurable canvas, Preview, Export to GIF and much more.
          </p>
        </div>
        <div
          className="landing-page__redactor-screenshot"
        >
          <img
            alt="Redactor screenshot"
            src="test-file-stub"
          />
        </div>
      </div>
      <div
        className="landing-page__right-section"
      >
        <div
          className="landing-page__examples-section"
        >
          <h3>
            Examples
          </h3>
          <p>
            Here are some of the examples of animations that were created in this app.
          </p>
          <div
            className="landing-page__example"
          >
            <img
              alt="gif example"
              src="test-file-stub"
            />
            <img
              alt="gif example"
              src="test-file-stub"
            />
            <img
              alt="gif example"
              src="test-file-stub"
            />
          </div>
        </div>
        <button
          className="landing-page__main-button"
          onClick={[Function]}
          type="button"
        >
          <span>
            Go to app
          </span>
        </button>
      </div>
    </div>
  </main>,
  <footer
    className="footer"
  >
    <div
      className="footer__content"
    >
      <p>
        Developed with
        <span
          aria-label="red heart"
          className="footer__heart"
          role="img"
        >
          ❤️
        </span>
        by
        <a
          href="https://github.com/uniorunr"
          rel="noopener noreferrer"
          target="_blank"
        >
          uniorunr
        </a>
      </p>
    </div>
  </footer>,
]
`;
